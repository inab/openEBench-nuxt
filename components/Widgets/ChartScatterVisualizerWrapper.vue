<template>
	<div
		:data-id="id"
		class="benchmarkingChart_scatter"
		:data-api-url="apiUrl"
		toTable="true"
		data-mode="openebench"
	/>
</template>

<script>
// eslint-disable-next-line camelcase
import { load_scatter_visualization } from '@inb/oeb-chart-scatter';

export default {
	props: {
		id: {
			type: String,
			required: true,
		},
	},
	data() {
		return {
			apiUrl: this.$config
				? this.$config.SCIENTIFIC_SERVICE_URL + '/graphql'
				: 'https://dev-openebench.bsc.es/api/scientific/graphql',
		};
	},
	mounted() {
		this.loadVisualization();
	},
	methods: {
		loadVisualization() {
			try {
				load_scatter_visualization();
			} catch (error) {}
		},
	},
};
</script>

<style lang="scss">
.benchmarkingTable_scatter {
	background-color: #fff;
	width: 28vw;
}

.benchmarkingSVG {
	width: 69vw !important;
}
</style>
>
