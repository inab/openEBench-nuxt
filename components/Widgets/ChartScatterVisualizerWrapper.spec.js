import { shallowMount } from '@vue/test-utils';
import ChartScatterVisualizerWrapper from './ChartScatterVisualizerWrapper.vue';

jest.mock('@inb/oeb-chart-scatter');
// eslint-disable-next-line
import { load_scatter_visualization } from '@inb/oeb-chart-scatter';

const factory = () => {
	return shallowMount(ChartScatterVisualizerWrapper, {
		...createComponentMocks({}),
		propsData,
	});
};

const propsData = { id: 'someid' };

describe('ChartScatterVisualizerWrapper', () => {
	afterEach(() => {
		jest.clearAllMocks();
	});

	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('is calls load_scatter_visualization after mount', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
		expect(load_scatter_visualization).toHaveBeenCalled();
	});

	it('sets the apiUrl according to SCIENTIFIC_SERVICE_URL env var', () => {
		const wrapper = factory();
		expect(wrapper.vm.$data.apiUrl).toBe(
			'https://jest-openebench.bsc.es/api/scientific/graphql'
		);
	});
});
