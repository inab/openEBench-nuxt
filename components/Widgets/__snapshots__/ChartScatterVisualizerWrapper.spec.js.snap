// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartScatterVisualizerWrapper should match snapshot 1`] = `"<div data-id=\\"someid\\" data-api-url=\\"https://jest-openebench.bsc.es/api/scientific/graphql\\" totable=\\"true\\" data-mode=\\"openebench\\" class=\\"benchmarkingChart_scatter\\"></div>"`;
